object_test.go

```diff
@@ -665,64 +665,156 @@ func TestObject_InList(t *testing.T) {
 }
 
 func TestObject_ContainsKey(t *testing.T) {
-	reporter := newMockReporter(t)
-
-	value := NewObject(reporter, map[string]interface{}{"foo": 123, "bar": ""})
-
-	value.ContainsKey("foo")
-	value.chain.assert(t, success)
-	value.chain.clear()
-
-	value.NotContainsKey("foo")
-	value.chain.assert(t, failure)
-	value.chain.clear()
-
-	value.ContainsKey("bar")
-	value.chain.assert(t, success)
-	value.chain.clear()
-
-	value.NotContainsKey("bar")
-	value.chain.assert(t, failure)
-	value.chain.clear()
-
-	value.ContainsKey("BAR")
-	value.chain.assert(t, failure)
-	value.chain.clear()
+	testObj := map[string]interface{}{"foo": 123, "bar": ""}
+	const (
+		contains    = iota
+		notContains = iota
+	)
+
+	cases := []struct {
+		name      string
+		object    map[string]interface{}
+		key       string
+		assertion uint
+		wantEqual chainResult
+	}{
+		{
+			name:      "1. correct key value, contains assertion",
+			object:    testObj,
+			key:       "foo",
+			assertion: contains,
+			wantEqual: success,
+		},
+		{
+			name:      "1. correct key value, not contains assertion",
+			object:    testObj,
+			key:       "foo",
+			assertion: notContains,
+			wantEqual: failure,
+		},
+		{
+			name:      "2. correct key value, contains assertion",
+			object:    testObj,
+			key:       "bar",
+			assertion: contains,
+			wantEqual: success,
+		},
+		{
+			name:      "2. correct key value, not contains assertion",
+			object:    testObj,
+			key:       "bar",
+			assertion: notContains,
+			wantEqual: failure,
+		},
+		{
+			name:      "3. wrong key value, contains assertion",
+			object:    testObj,
+			key:       "BAR",
+			assertion: contains,
+			wantEqual: failure,
+		},
+		{
+			name:      "3. wrong key value, not contains assertion",
+			object:    testObj,
+			key:       "BAR",
+			assertion: notContains,
+			wantEqual: success,
+		},
+	}
 
-	value.NotContainsKey("BAR")
-	value.chain.assert(t, success)
-	value.chain.clear()
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			reporter := newMockReporter(t)
+			value := NewObject(reporter, tc.object)
+
+			if tc.assertion == contains {
+				value.ContainsKey(tc.key)
+				value.chain.assert(t, tc.wantEqual)
+				value.chain.clear()
```

Comment:
Could you please use the same approach as, for example, [here](https://github.com/gavv/httpexpect/blob/dad9859e10c0a94862136849954adb6bcfeb1ac8/number_test.go#L824-L825)?

Like:

```go
		NewObject(...).ContainsKey(...).
			chain.assert(...)
``` 

With this approach, each check uses a fresh value, instead of clearing chain and reusing old value. It's more clear and safe because there is no shared state.

Reply:
Same for other tests in this PR.

Reply:
Could you also apply this approach to subtests of the modified tests, which you didn't convert to table driven form?

(as mentioned in the last paragraph of the issue)

They include:

* [ ] TestObject_ContainsValue/struct
* [ ] TestObject_ContainsValue/canonization
* [ ] TestObject_ContainsValue/invalid argument
* [ ] TestObject_ContainsSubset/struct
* [ ] TestObject_ContainsSubset/canonization
* [ ] TestObject_HasValue/struct
* [ ] TestObject_HasValue/canonization